### DISCUSSION TITLE
Third-party packages conflicts with  trafficCop

### DISCUSSION THREAD

**Discussion by beshoo** [laravel/nova-issues#5639] [Q&A]
**Summary**
When using the **serenysoft/nova-permissions** package along with Laravel Nova, updating resources inside the Nova dashboard fails with the error message: "Another user has updated this resource since this page was loaded. Please refresh the page and try again." 

This issue occurs browser time zones are NOT in sync with the server `(That is what Nova support responds to me).`

**Steps to Reproduce**
Install the `serenysoft/nova-permissions` package in a Laravel Nova project.
Create a resource and attach permission to it using the `serenysoft/nova-permissions` package.
Attempt to update the resource permission inside the Nova dashboard. 

The update will fail with the error message: "**Another user has updated this resource since this page was loaded. Please refresh the page and try again**." 

**Expected Behavior**
The update should be successful without any error messages.

**Actual Behavior**
The update fails with the error message: "Another user has updated this resource since this page was loaded. Please refresh the page and try again."

**Environment**
Laravel Version:10
Nova Version: 4.x
`serenysoft/nova-permissions Package`
Server Timezone:  `UTC`
MySQL Timezone: `UTC`
Laravel Timezone in app.php: `UTC`
Browser Timezne: `Unknown`

**Workaround**
Disabling the trafficCop feature by setting public static `$trafficCop = false;` in **app/Nova/Resources.php** resolves the issue for resources defined in the app/Nova directory. 

However, this solution does not work for resources defined in the `serenysoft/nova-permissions `package, which has its own **vendor/sereny/nova-permissions/src/Nova/Resource.php** file that extends the main **vendor/laravel/nova/src/Resource.php** file. which means it ignores "**app/Nova/Resources.php**" altogether.
Any plugin may have this method of OOP and we can not follow all plugins!

**Despite updating the plugin to solve the problem but any time I do **composer update** , all my modifications will be lost.**

**Expected Solution**
I suggest adding a configuration flag in Laravel Nova's configuration file that can be used to disable the trafficCop feature globally. This will allow users to use third-party packages like `serenysoft/nova-permissions` without encountering conflicts with the trafficCop feature.

**Comment by crynobone** [comment]
You can just extend the relevant Resource if you have this requirement:

```php
<?php

namespace App\Nova;

use Laravel\Nova\Http\Requests\NovaRequest;
use Sereny\NovaPermissions\Nova\Resource as NovaResource;

abstract class Resource extends NovaResource
{
    /**
     * Indicates whether Nova should check for modifications between viewing and updating a resource.
     *
     * @var bool
     */
    public static $trafficCop = false;
```

**Reply by beshoo** [reply]
Yes but why make it so hard for the user?

**Reply by crynobone** [reply]
The package made it hard to user, they should make it as a `trait` instead.

**Reply by beshoo** [reply]
Thank you for your reply, but I did not follow you, I need to update the plugin files? 
Where would I do this extend...

![image](https://github.com/laravel/nova-issues/assets/4635115/f7a886a5-40f7-4260-9a52-5cd445fa252b)

It does not work this way!

**Reply by crynobone** [reply]
Does your `User` and `Attended` resource extends `App\Nova\Resource`?

**Comment by beshoo** [comment]
![image](https://github.com/laravel/nova-issues/assets/4635115/1d087292-d541-4055-b432-d2af592619ea)
![image](https://github.com/laravel/nova-issues/assets/4635115/ff174f68-484c-4fd2-b810-11754e5382e8)

Please note that I don't have a problem with the user and attended resource since I add 

![image](https://github.com/laravel/nova-issues/assets/4635115/d1398df7-cabb-4988-bb98-d86f7cb0e232)

The problem with the vendor's third-party plugins!

**Reply by crynobone** [reply]
> The problem with the vendor's third-party plugins!

You should report this to the vendor's third-party plugin.
